locust_tasks/locustfile.py

```diff
@@ -166,49 +189,163 @@ def correct_address(self):
 The simulated user steps through the pages one by one. They don't go down any of the 
 correction/error paths as this doesn't trigger any significant server side work.
 """
-class request_new_code(SequentialTaskSet):
+class RequestNewCodeSMS(SequentialTaskSet):
+    """
+    Class to represent a user requesting a new UAC, which is to be sent by SMS.
+    """
 
-    # All users arrive at the start page
     @task(1)
     def start_page(self):
-        self.client.get("/en/start/")
+        """
+        GET Start page
+        """
+        self.case = get_next_case()
+        with self.client.get('/en/start/', catch_response=True) as response:
+            verify_response('RequestUacSms-Start', self, response, 200, Page.START)
         
     @task(2)
-    def request_new_access_code(self):
-        self.client.get("/request-access-code")
+    def enter_postcode(self):
+        """
+        POST postcode
+        """
+        with self.client.post("/en/requests/access-code/enter-address/", {
+            'form-enter-address-postcode': self.case['postcode']
+        }, catch_response=True) as response:
+            self.address_to_select = extractAddress(response, self.case["uprn"])
+            verify_response('RequestUacSms-EnterAddress', self, response, 200, Page.SELECT_ADDRESS,
+                            self.case["postcode"])
 
     @task(3)
     def select_address(self):
-        # This code should arguable select one of the available addresses but running 
-        # with a fixed address doesn't seem to affect the success of the test
-        self.client.post("/request-access-code/select-address", {
-            'request-address-select': "{'uprn': '10023122452', 'address': hardcoded_address}"
-        })
+        """
+        POST uprn and whole address extracted as JSON from the HTML in previous task
+        """
+        logger.info("Address selected: " + self.address_to_select)
```

Comment:
Can't have info logging

Reply:
I'm not sure what the rules are regarding logging?

Reply:
You can have it during development but it's just not sensible for production runs. We will be doing 1000 requests per second, so even if some small fraction of them are producing output it'll be a mess.
It needs to run like Unix commands: only outputting anything when it's not working.

Reply:
got it, thanks

Reply:
I've deleted it now